File: backend/src/main/java/com/example/ecommerce/service/customer/CustomerService.java
package com.example.ecommerce.service.customer;

import java.util.List;

import org.springframework.http.ResponseEntity;

import com.example.ecommerce.dto.CartItemsDTO;
import com.example.ecommerce.dto.OrderDTO;
import com.example.ecommerce.dto.PlaceOrderDTO;
import com.example.ecommerce.dto.ProductDTO;

public interface CustomerService {
	
	List<ProductDTO> getAllProducts();

	List<ProductDTO> searchProductByTitle(String title);
	
	ResponseEntity<?> addProductCart(CartItemsDTO cartItemsDTO);
	
	OrderDTO getCartByUserid(Long userid);
	
	OrderDTO addMinusOnProduct(Long userId, Long productId);
	
	OrderDTO addPlusOnProduct(Long userId, Long productId);
	
	OrderDTO placeOrder(PlaceOrderDTO placeOrderDTO);
	
	List<OrderDTO> getOrdersByUserId(Long userId);
}

File: frontend-angular/src/app/user/user-service/customer.service.ts
import { Injectable } from '@angular/core';
import { apiServer } from '../../apiServer';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { LocalstorageService } from '../../services/storage-service/localstorage.service';

const BASIC_URL = apiServer['serverUrl'];

@Injectable({
  providedIn: 'root'
})
export class CustomerService {


  constructor(private http: HttpClient) { }

  getAllProducts(): Observable<any>{
    return this.http.get<[]>(BASIC_URL+"api/customer/products");
  }

  searchProductById(title: String):Observable<any>{
    return this.http.get<[]>(BASIC_URL+"api/customer/product/search/"+title);
  }

  addProductToCart(productId:number):Observable<any>{
    let cartDTO ={
      productId:productId,
      userId:LocalstorageService.getUser()
    }
    return this.http.post<[]>(BASIC_URL+ "api/customer/cart",cartDTO)
  }

  getCartByUserId(): Observable<any>{
    return this.http.get<[]>(BASIC_URL+"api/customer/cart/" + LocalstorageService.getUser());
  }

  decreaseQuantityProduct(productId:number):Observable<any>{
    return this.http.get(BASIC_URL + `api/customer/${LocalstorageService.getUser()}/discount/${productId}`);
  }

  incrementQuantityProduct(productId:number):Observable<any>{
    return this.http.get(BASIC_URL + `api/customer/${LocalstorageService.getUser()}/add/${productId}`);
  
  }

  placeOrder(placeOrderDTO:any): Observable<any>{
    placeOrderDTO.userId= LocalstorageService.getUser();
    return this.http.post<[]>(BASIC_URL+"api/customer/placeOrder",placeOrderDTO);
  }

  getOrdersByUserId(): Observable<any>{
    return this.http.get(BASIC_URL+"api/customer/orders/" + LocalstorageService.getUser());
  }
}
